src/cascadia/LocalTests_TerminalApp/TabTests.cpp

```diff
@@ -868,13 +868,39 @@ namespace TerminalAppLocalTests
         VERIFY_ARE_EQUAL(4u, page->_mruTabs.Size());
 
         Log::Comment(L"give alphabetical names to all switch tab actions");
-        RunOnUIThread([&page]() {
+        TestOnUIThread([&page]() {
             page->_GetTerminalTabImpl(page->_tabs.GetAt(0))->Title(L"a");
+        });
+        TestOnUIThread([&page]() {
             page->_GetTerminalTabImpl(page->_tabs.GetAt(1))->Title(L"b");
+        });
+        TestOnUIThread([&page]() {
             page->_GetTerminalTabImpl(page->_tabs.GetAt(2))->Title(L"c");
+        });
+        TestOnUIThread([&page]() {
             page->_GetTerminalTabImpl(page->_tabs.GetAt(3))->Title(L"d");
         });
 
+        // Mysteriously, setting the title isn't instantaneous. If you don't
+        // sleep here, then it's very possible that the checks below to sanity
+        // check the titles _will fail_.
+        Log::Comment(L"Sleep to let events propagate");
+        Sleep(500);

```

Comment:
Instead of a sleep which could vary in length, can we make an event or mutex or something above... dispatch it on a `TestOnUIThread` element after all the other tests in a lambda that does nothing but set/trigger it, and then wait here on it until it's triggered as the last queued packet on the UI thread? That would likely assure that the other events have propagated in a more reliable manner?

Reply:
I would love to do that. Unfortunately, the event I'm waiting for here is "let XAML do it's thing". I'm not really sure there's a specific event I can wait for in this case 😕